presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -291,7 +292,7 @@ public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, N
         public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
         {
             // A grouping() is preceded by a GroupId node when the source node is projecting a groupid symbol
-            if (!source.getOutputSymbols().contains(new Symbol("groupid")) && node.getName().toString().equals(GROUPING)) {
+            if (!source.getOutputSymbols().stream().anyMatch(symbol -> symbol.getName().contains(GROUPID_SYMBOL_HINT)) && node.getName().toString().equals(GROUPING)) {
```

Comment:
I think this might be still incorrect. Take a look at example:
`SELECT GROUPING(...) FROM (SELECT ... FROM (SELECT ... GROUP BY ...))`.
Then there will be `groupId_XX` symbol, however it won't be directly related to top `GROUPING` function call.

So my concern is that you match `groupId` symbol that "isn't" for the `GRUPING()` function